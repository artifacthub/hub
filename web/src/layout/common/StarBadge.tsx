import classnames from 'classnames';
import { isNumber, isUndefined } from 'lodash';
import React from 'react';
import { FaStar } from 'react-icons/fa';

import prettifyNumber from '../../utils/prettifyNumber';
import styles from './StarBadge.module.css';

interface Props {
  starsNumber?: number | any;
  className?: string;
  size?: 'xs';
}

const StarBadge = (props: Props) => {
  if (isUndefined(props.starsNumber) || !isNumber(props.starsNumber)) return null;
  return (
    <div
      data-testid="starBadge"
      className={classnames('badge badge-pill badge-light', styles.badge, props.className, {
        [styles[`size-${props.size}`]]: !isUndefined(props.size),
      })}
    >
      <div className="d-flex align-items-center">
        <FaStar className={`mr-1 ${styles.icon}`} />
        <div>{prettifyNumber(props.starsNumber)}</div>
      </div>
    </div>
  );
};

export default StarBadge;
