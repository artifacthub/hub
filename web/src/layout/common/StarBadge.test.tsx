import { render } from '@testing-library/react';
import React from 'react';

import StarBadge from './StarBadge';

describe('StarBadge', () => {
  it('creates snapshot', () => {
    const { asFragment } = render(<StarBadge starsNumber={1} />);
    expect(asFragment).toMatchSnapshot();
  });

  it('renders proper content', () => {
    const { getByTestId, getByText } = render(<StarBadge starsNumber={3} />);
    expect(getByTestId('starBadge')).toBeInTheDocument();
    expect(getByText(3)).toBeInTheDocument();
  });

  it('renders xs size', () => {
    const { getByTestId } = render(<StarBadge starsNumber={1} size="xs" />);
    const el = getByTestId('starBadge');
    expect(el).toHaveClass('size-xs');
  });

  describe('does not render component', () => {
    it('when stars number is undefined', () => {
      const { container } = render(<StarBadge />);
      expect(container).toBeEmptyDOMElement();
    });

    it('when stars is not a number', () => {
      const { container } = render(<StarBadge starsNumber="7" />);
      expect(container).toBeEmptyDOMElement();
    });
  });
});
