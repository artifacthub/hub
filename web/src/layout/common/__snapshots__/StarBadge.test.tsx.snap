// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarBadge creates snapshot 1`] = `[Function]`;
